coordinator/tests/chain.rs

```diff
@@ -95,7 +95,7 @@ static ONCE: OnceCell<Mutex<SharedState>> = OnceCell::const_new();
 
 async fn get_shared_state() -> &'static Mutex<SharedState> {
     ONCE.get_or_init(|| async {
-        let shared_state = SharedState::from_env().await;
+        let shared_state = SharedState::from_env_for_tests().await;
```

Comment:
Can we keep `from_env`? Or is there a good reason not to?

Reply:
Related: If you take a look at `shared_state.rs` the `ro` field can be actually replaced with the config. And then just use it everywhere

Reply:
@pinkiebell the ro: RoState field has more stuf than config, and config has things that are not in RoState. Do you want me to put config as a field of RoState and remove the fields in RoState that are in Config?

Reply:
@pinkiebell I kept from_env because the testing environment do not need some fields in Config that in our clap are mandatory.

Reply:
I see. What about just adding a `config: ...` field then?

Reply:
Btw you can add a `pub fn from_env` to the config struct and do something like this: `MyConfig::parse_from(std::env::args().skip(usize::MAX));` that basically skips all cmdline arguments and only parses from env.